modules/thermal_hydraulics/src/base/Simulation.C

```diff
@@ -564,6 +572,63 @@ Simulation::addComponentScalarIC(const VariableName & var_name, const std::vecto
   addSimInitialCondition(class_name, genName(var_name, "ic"), params);
 }
 
+std::vector<VariableName>
+Simulation::sortAddedComponentVariables() const
+{
+  // Check that no index in order map is used more than once.
+  // Also, convert the map to a vector pairs to be sorted.
+  std::vector<int> indices;
+  std::vector<std::pair<VariableName, int>> registered_var_index_pairs;
+  for (const auto & var_and_index : _component_variable_order_map)
+  {
+    registered_var_index_pairs.push_back(var_and_index);
+
+    const auto ind = var_and_index.second;
+    if (std::find(indices.begin(), indices.end(), ind) != indices.end())
+      mooseError("The index ", ind, " was used for multiple component variables.");
+    else
+      indices.push_back(ind);
```

Comment:
if you used a set for indices you could check whether insertion worked. a little faster (N log N)
N searches in vector is N2

fine for small numbers here